src/odemis/driver/test/ueye_test.py

```diff
@@ -97,5 +98,47 @@ def setUp(self):
         VirtualTestCam.setUp(self)
 
 
+class TestUEyeCroppingTranspose(VirtualTestCam, unittest.TestCase):
+    camera_type = CLASS
+    camera_kwargs = KWARGS
+
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW:
+            raise unittest.SkipTest('No camera HW present. Skipping tests.')
+
+        cls.camera_kwargs['max_res'] = (1024, 768)
+        # apply a transpose of 90 CW and mirror ax-0
+        cls.camera_kwargs['transp'] = [-2, 1]
+        super(TestUEyeCroppingTranspose, cls).setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        if TEST_NOHW:
+            return
+        super(TestUEyeCroppingTranspose, cls).tearDownClass()
+
+    def setUp(self):
+        if TEST_NOHW:
+            self.skipTest("No simulator available")
+
+        VirtualTestCam.setUp(self)
+
+    def test_resolution(self):
+        # check if the resolution is set correctly in the initialization of the camera class
+        cam_resolution = self.camera._sensor_res
+        trans_resolution = self.camera._metadata[model.MD_SENSOR_SIZE]
+
+        # check if the resolution is not the same own cam res should be higher
+        self.assertNotEqual(self.camera_kwargs['max_res'], cam_resolution)
+        # check if the transpose parameter is applied and axes are switched
+        self.assertEqual((768, 1024), trans_resolution)
+        # check is the AOI is set with the right x, y, height and width values
+        self.assertEqual(self.camera.rect_aoi.s32X, 128)
+        self.assertEqual(self.camera.rect_aoi.s32Y, 128)
+        self.assertEqual(self.camera.rect_aoi.s32Width, 1024)
+        self.assertEqual(self.camera.rect_aoi.s32Height, 768)
```

Comment:
No need to check that. It's something private to the driver (and I've just asked you to remove it ;-) ).
However, you should acquire some images, and check they are the expected resolution.
See `self.camera.data.get()`.
Afterwards, set the binning to (for example) "2, 4", and check that the resolution of the returned image is X/2 and Y/4.